src/categorical_algebra/FreeDiagrams.jl

```diff
@@ -175,6 +176,93 @@ Base.lastindex(para::ParallelMorphisms) = lastindex(para.homs)
 
 allequal(xs::AbstractVector) = all(isequal(x, xs[1]) for x in xs[2:end])
 
+# SquareDiagrams form a Double category
+#######################################
+"""    SquareDiagram(top, bottom, left, right)
+
+creates a square diagram in a category, which forms the 2-cells of the double category Sq(C).
+The four 1-cells are given in top, bottom, left, right order, to match the GAT of a double category.
+"""
+@auto_hash_equals struct SquareDiagram{Ob,Homs<:AbstractVector} <:
+    FixedShapeFreeDiagram{Ob}
+  corners::Vector{Ob}
```

Comment:
Why not a `StaticVector` of length 4?

Reply:
Or named tuples with descriptive field names, or even just fields? IIRC, @jpfairbanks said in Zulip that he didn't want to always work with a conventional order of corners/sides. I'd agree with that.